modules/exceptions.py

```diff
@@ -0,0 +1,2 @@
+class ApiResponseError(Exception):
```

Comment:
You collapsed this file down to one single type? I do think it's a good idea to keep a few different types, I was just suggesting that you have a unified base-type so we could check for any type of error coming from this layer. e.g.:

``` python
class ApiError(Exception):
    pass

class ApiTimeoutError(ApiError):
    pass

class ApiResponseDecodeError(ApiError):
    pass

class ApiResponseErrorList(ApiError):
    def __init__(self, message, errors):
        self.errors = errors
        return super().__init__(message)
```

Then users of `api.py` can be general or specific when handling errors:

``` python
try:
    api.do_something()
except ApiTimeoutError:
    print('API was super-slow')
except APIError:
    print('API had some other error')
```